src/JuliennedArrays.jl

```diff
@@ -115,13 +118,14 @@ julia> size(first(larger_slices))
 (5,)
 ```
 """
-Slices(whole::AbstractArray, alongs::TypedBool...) = Slices{
-    typeof(@inbounds view(whole, map(axis_or_1, alongs, axes(whole))...)),
-    length(getindex_unrolled(alongs, map(not, alongs))),
-}(
-    whole,
-    alongs,
-)
+function Slices(whole::AbstractArray, alongs::TypedBool...)
+    length(alongs) <= ndims(whole) || throw(ArgumentError("$(length(alongs)) dimensions are specified, expected to be <= $(ndims(whole))"))
```

Comment:
In general, I think it would be nice for these kind of checks to be done at compile-time, not run-time. So for example, if you used `===` here, then constant propagation could figure this out